### PR TITLE
refactor!: rework ci & code cleanup

### PR DESCRIPTION
**Pull request by davidkna**
This PR includes:

- release-please and cargo publishing
- (Build) testing for more platforms (including FreeBSD, musl and ARM)
- Upgraded dependencies
- Depdendabot
- Clippy fixes
- removal of disabled clippy lints
- removal of deprecated std-library item (`mem::uninitialized`)
- add audit-check to CI

Marked as a breaking change because this includes the non-exhaustive removal from #2.

@matchai You will need to add secrets for cargo and Dependabot.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+version: 2
+updates:
+- package-ecosystem: cargo
+  directory: "/"
+  schedule:
+    interval: daily
+  open-pull-requests-limit: 10
+  labels:
+  - "\U0001F4E6 dependencies"
```
Comment (matchai) [inline .github/dependabot.yml:9]: I went ahead and created this label, just in case it's needed.

File: src/types/state.rs [hunk 2]
```diff
@@ -13,10 +13,6 @@ pub enum State {
     Discharging,
     Empty,
     Full,
-
-    // Awaiting for https://github.com/rust-lang/rust/issues/44109
-    #[doc(hidden)]
-    __Nonexhaustive,
```
Comment (matchai) [inline src/types/state.rs:19]: Did you intentionally not add `#[non_exhaustive]` here?
I'm fine either way – just wanted to make sure it was on purpose.

**Review by matchai — APPROVED** [review]
LGTM! 👍 
I went ahead and added the required secrets.

File: src/types/state.rs (hunk 2 above)
Comment (davidkna) [reply src/types/state.rs:19]: It's intentional. This matches the changes discussed in #2.

It looks like the library consumers that recently hopped-on would also want to handle additional enum values explicitly as they are added.

**Comment by davidkna** [comment]
Do we also want to set the main branch name to main as part of this, as a trial for the main repo?

**Comment by matchai** [comment]
Yes, sounds like a good idea! 👍

**Comment by chipbuster** [comment]
@davidkna That looked like a big ol ball of work, thanks for getting it taken care of.

**Pull request merged**